portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java

```diff
@@ -31,14 +31,22 @@
 public class CommentsServlet extends HttpServlet {
   private static final String JSON_CONTENT_TYPE = "application/json;";
   private static final String COMMENTS_PAGE_URL = "/comments-page.html";
+  private static final int MAX_COMMENT_DEFAULT = 20;
   DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
   CommentDatabase commentDatabase = new CommentDatabase(datastore);
 
   /* responds with a JSON string containing authors and comments*/
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Query datastore for all comments
-    ArrayList<Comment> comments = commentDatabase.getCommentsAsArrayList();
+    int maxComments = MAX_COMMENT_DEFAULT;
+    try {
+        maxComments = getMaxCommentsToDisplay(request);
+    } catch (IllegalArgumentException e) {
```

Comment:
You probably need to catch NumberFormatException here too

Reply:
I believe NumberFormatException is a subclass of IllegalArgumentException